### DISCUSSION TITLE
Is there a way to use `post-render` when rendering to `gfm` format?

### DISCUSSION THREAD

**Discussion by danielvartan** [quarto-dev/quarto-cli#10718] [Q&A] [answered]
### Description

Hi there, 👋 

I would like to run an R script after rendering a Quarto document with a GitHub README file (`format: gfm`). Is that possible?

```qmd
---
title: "Test"
format: gfm
post-render: "R/_post-render.R"

editor: source
---

## Test

Phasellus laoreet lorem vel dolor tempus vehicula. Idque Caesaris facere
voluntate liceret: sese habere. Ab illo tempore, ab est sed immemorabili.
Mercedem aut nummos unde unde extricat, amaras. Praeterea iter est quasdam res
quas ex communi.
```

**Comment by mcanouil** [comment]
Post-render is a project option unrelated to format. All projects options go to `_quarto.yml` as documented.

Where did you see that you could set this in a document?

https://quarto.org/docs/projects/scripts.html#pre-and-post-render

**Reply by danielvartan** [reply]
> Where did you see that you could set this in a document?

I didn't. That's why I asked.

It's just an README file generator. What could be the easiest way to just render a GitHub `README.md` and include a post-render script in the process (without making a mess)?

**Reply by mcanouil** [reply]
Exactly as I said previously. This works as a project level option.

**Reply by danielvartan** [reply, marked as answer]
Yes, but it's not a simple solution.

I found a way, as you mentioned, by creating a project file (`_quarto.yml`) like the one below.

```qmd
editor: source

project:
  output-dir: "."
  render: [README.qmd]
  post-render: "R/_post-render.R"

title: "Test"

format: gfm
```

Thank you for your assistance.

**Reply by mcanouil** [reply]
> Yes, but it's not a simple solution

I don't know how it's not simple to set a project configuration file.
It's how it works and how it is documented.
